code/modules/mining/equipment_locker.dm

```diff
@@ -507,8 +507,8 @@
 	if(istype(proj_turf, /turf/simulated/mineral))
 		var/turf/simulated/mineral/M = proj_turf
 		playsound(src,'sound/weapons/resonator_blast.ogg',50,1)
-		M.gets_drilled()
 		spawn(5)
+			M.gets_drilled(creator)
```

Comment:
so the resonance effect is given the proper creator value. On line 478 the resonance is created with New and on the next line it's given its creator value. So the spawn makes sure the creator argument of this gets_drilled is the attacker instead of null. And since the resonator weapon's effect has the particularity of lasting some time before disappearing, I don't think the delayed action is unnatural.